wurst/systems/boards/PlayerScore.wurst

```diff
@@ -275,19 +287,35 @@ init
                     and target.isTroll()
                     and playerScores.has(attackerOwner)
                     playerScores.get(attackerOwner).addDamage(dmgAmount.toInt())
-                    playerScores.get(attackerOwner.getTribe().getNonMember()).addDamage(dmgAmount.toInt())
+
+                    if gameConfig.isTribeOwnsBuildingsEnabled()
+                        playerScores.get(attackerOwner.getTribe().getNonMember()).addDamage(dmgAmount.toInt())
 
             // Register healing amount if player heals an ally
             onUnitHealed() ->
                 let instance = getHealingInstance()
                 let caster = instance.getCaster()
                 let target = instance.getTarget()
                 let healerOwner = caster.getOwner()
+                let healingAmount = instance.getTrueAmount()
 
-                if caster != target
-                    let healAmount = instance.getAmount()
-                    playerScores.get(healerOwner).addHeal(healAmount.toInt())
-                    playerScores.get(healerOwner.getTribe().getNonMember()).addHeal(healAmount.toInt())
+                // Do not take pet healing into account, atleast for now
+                if target.isTroll()
+                    if healerOwner != target.getOwner()
+                        playerScores.get(healerOwner).addAllyHealing(healingAmount.toInt())
+
+                        if gameConfig.isTribeOwnsBuildingsEnabled()
+                            playerScores.get(healerOwner.getTribe().getNonMember()).addAllyHealing(healingAmount.toInt())
+
+                    // Do not include self healing consumable items
+                    // I have mixed opinion on this, we should differentiate healing done from ability/item
+                    // This we can "quantify" how much cooked meat priest can actually save through healing spells
+                    // Still, I think having the healing done from meat could also be useful
+                    else if healerOwner == target.getOwner() and instance.getHealingType() != HealingType.ITEM
+                        playerScores.get(healerOwner).addSelfHealing(healingAmount.toInt())
+
+                        if gameConfig.isTribeOwnsBuildingsEnabled()
```

Comment:
What is the point of this line? Just so that the tribe is equal to the sum of the members instead of `0`? Why is that necessary?

Reply:
> 
> 
> How does this change the behavior for Cloak of Healing like the changelog mentions? It looks both types of healing worked before the change.

Cloak Of Healing healing instances do not register the caster, I simply didn't put the caster as parameter of the constructor, when no caster is specified, the instance [set the caster as target.](https://github.com/island-troll-tribes/island-troll-tribes/blob/ffde7d8cf80fd3df5dc807eea06919193635c24a/wurst/lib/HealingSystem.wurst#L44)

My condition to detect "healing done to allies" was `if caster != target`, I could have fixed healing cloak healing by specifying the caster in the constructor, however I also needed to handle healing casted by dummyCaster, e.g, sage light gate, so I changed the condition to detect the healing instance owner instead.

Note: OK I wrote this to explain how I fixed it, just to realize, that I actually didn't fix it, healing cloak instances need the caster unit, otherwise they will set the caster as target and healingOwner will equal healingOwner, shaking my head.

Reply:
> 
> 
> What is the point of this line? Just so that the tribe is equal to the sum of the members instead of `0`? Why is that necessary?

playerScores lookup the instances using the player variable, I assumed the player score would work on par with tb, as tb assign a playerId to the tribe, I could consider the tribe as a player and create a player score instance for it. 

If tb is disabled, there's no row displayed for tribe computed score.

The issue here, is that I do not create a player score tribe if tb is disabled, yet I try to access it through playerScores IterableMap, which result in a nullPointerException, big mistake.

Adding those if conditions prevent the "illegal access" if that's the right expression, but I can seem that the design is flawed.